packages/utils/src/numbers/decimals.ts

```diff
@@ -5,17 +5,22 @@ export function toDecimal(
   decimals: number,
   maxDecimal = 6,
   roundType?: RoundingMode,
-): number {
+): Big {
   const dividend = Big(number.toString().replace(/[^0-9]/g, ''));
   const divisor = Big(10).pow(decimals);
   const result = dividend.div(divisor).round(maxDecimal, roundType);
 
-  return result.toNumber();
+  return result;
 }
 
 export function toBigInt(amount: string | number, decimals: number): bigint {
   const multiplier = Big(10).pow(decimals);
   const result = Big(amount).mul(multiplier);
 
-  return BigInt(result.toFixed());
+  return BigInt(result.toFixed(0, 0));
```

Comment:
If the method receives a number with more decimals than allowed, it throws an uncontrolled error when trying to create a bigint from a number with decimals. By passing` toFixed(0,0)` we ensure that the number being converted doesn't have decimals, effectively truncating the decimals on the right. 

Another approach is using the `hasDecimalOverflow` function to throw a controlled error but I feel like the first option is simpler and will work for most cases. We can debate this though

Reply:
I think `toFixed(0,0)` is fine and should work.